test/e2e/flowaggregator_test.go

```diff
@@ -430,20 +431,42 @@ func testHelper(t *testing.T, data *TestData, podAIPs, podBIPs, podCIPs, podDIPs
 	// LocalServiceAccess tests the case, where Pod and Service are deployed on the same Node and their flow information is exported as IPFIX flow records.
 	t.Run("LocalServiceAccess", func(t *testing.T) {
 		skipIfProxyDisabled(t, data)
-		if !isIPv6 {
-			checkRecordsForFlows(t, data, podAIPs.ipv4.String(), svcB.Spec.ClusterIP, isIPv6, true, true, false, false, checkBandwidth)
+		// In dual stack cluster, service IP can be assigned as different IP family from specified.
+		// In that case, source IP and destination IP will align with IP family of service IP.
+		isServiceIPv6 := net.ParseIP(svcB.Spec.ClusterIP).To4() == nil
+		if isServiceIPv6 != isIPv6 {
+			if isServiceIPv6 {
+				checkRecordsForFlows(t, data, podAIPs.ipv6.String(), svcB.Spec.ClusterIP, isServiceIPv6, true, true, false, false, checkBandwidth)
+			} else {
+				checkRecordsForFlows(t, data, podAIPs.ipv4.String(), svcB.Spec.ClusterIP, isServiceIPv6, true, true, false, false, checkBandwidth)
+			}
```

Comment:
Could we use `isServiceIPv6` if the cluster is dual-stack and`isIPv6` for other clusters? We can limit the code to one if-else block.